flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/MiniClusterITCase.java

```diff
@@ -337,6 +347,262 @@ public void testSchedulingAllAtOnce() throws Exception {
+
+		try (final MiniCluster miniCluster = new MiniCluster(cfg)) {
+			miniCluster.start();
+
+			final JobVertex sender1 = new JobVertex("Sender1");
+			sender1.setInvokableClass(CountDownLatchedSender.class);
+			sender1.setParallelism(parallelism);
+
+			final JobVertex sender2 = new JobVertex("Sender2");
+			sender2.setInvokableClass(CountDownLatchedSender.class);
+			sender2.setParallelism(parallelism);
+
+			final JobVertex receiver = new JobVertex("Receiver");
+			receiver.setInvokableClass(CountDownLatchedAgnosticBinaryReceiver.class);
+			receiver.setParallelism(parallelism);
+
+			receiver.connectNewDataSetAsInput(sender1, DistributionPattern.POINTWISE,
+				ResultPartitionType.PIPELINED);
+			receiver.connectNewDataSetAsInput(sender2, DistributionPattern.ALL_TO_ALL,
+				ResultPartitionType.PIPELINED);
+
+			final CountDownLatch countDownLatch = new CountDownLatch(parallelism);
+			CountDownLatchedSender.setLatch(countDownLatch);
+			CountDownLatchedAgnosticBinaryReceiver.setLatch(countDownLatch);
+
+			final SlotSharingGroup sharingGroup = new SlotSharingGroup(sender1.getID(), sender2.getID(), receiver.getID());
+			sender1.setSlotSharingGroup(sharingGroup);
+			sender2.setSlotSharingGroup(sharingGroup);
+			receiver.setSlotSharingGroup(sharingGroup);
+
+			final JobGraph jobGraph = new JobGraph("Bipartite job", sender1, sender2, receiver);
+
+			miniCluster.executeJobBlocking(jobGraph);
+		}
+	}
+
+	/**
+	 * A receiver that counts down the latch on running.
+	 */
+	public static class CountDownLatchedAgnosticBinaryReceiver extends AbstractInvokable {
+
+		private static CountDownLatch latch;
+
+		static void setLatch(CountDownLatch latch) {
+			CountDownLatchedAgnosticBinaryReceiver.latch = latch;
+		}
+
+		public CountDownLatchedAgnosticBinaryReceiver(Environment environment) {
+			super(environment);
+		}
+
+		@Override
+		public void invoke() throws Exception {
+			latch.countDown();
+
+			RecordReader<IntValue> reader1 = new RecordReader<>(
+				getEnvironment().getInputGate(0),
+				IntValue.class,
+				getEnvironment().getTaskManagerInfo().getTmpDirectories());
+
+			RecordReader<IntValue> reader2 = new RecordReader<>(
+				getEnvironment().getInputGate(1),
+				IntValue.class,
+				getEnvironment().getTaskManagerInfo().getTmpDirectories());
+
+			while (reader1.next() != null) { }
+			while (reader2.next() != null) { }
+		}
+	}
+
+	@Test
+	public void testSlotSharingForForwardJobWithFailedTaskManager() throws Exception {
+		final int parallelism = 20;
+		final int numOfTaskManagers = 2;
+
+		final Configuration configuration = getDefaultConfiguration();
+		configuration.setLong(JobManagerOptions.SLOT_REQUEST_TIMEOUT, 100L);
+
+		final MiniClusterConfiguration cfg = new MiniClusterConfiguration.Builder()
+			.setNumTaskManagers(numOfTaskManagers)
+			.setNumSlotsPerTaskManager(parallelism / numOfTaskManagers)
+			.setConfiguration(configuration)
+			.build();
+
+		try (final MiniCluster miniCluster = new MiniCluster(cfg)) {
```

Comment:
I think it would be a good idea to separate tests which modify the `MiniCluster` from those which don't. For the latter, the `MiniCluster` could be started for the test class instead for every test. This will speed up the execution.